subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/JavaLibraryPlugin.java

```diff
@@ -39,23 +42,45 @@
     @Override
     public void apply(final Project project) {
         project.getPlugins().apply("java");
+        final VersionInfo versionInfo = project.getRootProject().getExtensions().findByType(VersionInfo.class);
 
         final CopySpec license = project.copySpec(new Action<CopySpec>() {
             public void execute(CopySpec copy) {
                 copy.from(project.getRootDir()).include("LICENSE");
             }
         });
 
+        final File dependenciesFile = new File(project.getBuildDir(), "dependency-info.json");
+
+        Task createDependencyInfoFileTask = TaskMaker.task(project, "createDependencyInfoFile", CreateDependencyInfoFileTask.class, new Action<CreateDependencyInfoFileTask>() {
+            @Override
+            public void execute(CreateDependencyInfoFileTask task) {
+                task.setDescription("Creates file with resolved runtime dependencies.");
+                task.setOutputFile(dependenciesFile);
+                task.setConfiguration(project.getConfigurations().getByName("runtime"));
+                task.setProjectGroup(project.getGroup().toString());
```

Comment:
Is it safe to access project's group outside of the 'deferredConfiguration'? Typically we put it to deferredConfiguration to honor user setting of the group.

I've looked at our builds and I think that we need deferredConfiguration for group for single-project builds. Imagine this sequence of plugins:

```
apply plugin: 'org.shipkit.java'
apply plugin: 'java' // <- adds createDependencyInfoFileTask

group = 'foo' // <- group configured after createDependencyInfoFileTask was added
```

 This is needed for single module projects that apply 'org.shipkit.java' and 'java' plugin in root project.